src/gui/qgsowssourcewidget.cpp

```diff
@@ -0,0 +1,114 @@
+/***************************************************************************
+    qgsowssourcewidget.cpp
+     --------------------------------------
+    Date                 : November 2021
+    Copyright            : (C) 2021 by Samweli Mwakisambwe
+    Email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsowssourcewidget.h"
+#include "qgsproviderregistry.h"
+#include "qgsmapcanvas.h"
+
+#include <QNetworkRequest>
+
+
+QgsOWSSourceWidget::QgsOWSSourceWidget( const QString &providerKey, QWidget *parent )
+  : QgsProviderSourceWidget( parent )
+  , mProviderKey( providerKey )
+{
+  setupUi( this );
+
+  QgsCoordinateReferenceSystem destinationCrs;
+  QgsCoordinateReferenceSystem crs = QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:4326" ) );
+  mSpatialExtentBox->setOutputCrs( crs );
+
+  if ( mapCanvas() && mapCanvas()->mapSettings().destinationCrs().isValid() )
```

Comment:
I don't see how mapCanvas() would ever be non-nullptr here, in the constructor. There hasn't been any opportunity at this stage for anything to call setMapCanvas on the widget.

I think this needs to be handled in the same way as the source select widget -- make QgsProviderSourceWidget::setMapCanvas virtual, and override it in this class with this logic.

Reply:
Right!